challenge.py

```diff
@@ -0,0 +1,117 @@
+import os
+import re
+import numpy as np
+
+List_calls = "calls.txt"
+List_files = "list_files.txt"
+result_stat = "stat.txt"
+flag_file = 1
+flag_calls = 1
+def result(filename):
+  
+  #local variable declaration
+  deleted = 0 
+  added = 0 
+  region = 0 
+  
+  #Regular expression declaration 
+  regex_FileName = '^diff --git a.+[ ]b/(.+?)\n'
+  regex_Del = "^-(?!--).*"
+  regex_Add = "^\+(?!\+\+).*"
+  regex_region = "^@@ -\d+(?:,\d+)? \+\d+(?:,\d+)?\ @@[ ]?.*"
+  regex_functionCall = "([\w]+)(?=\()"
+  #catch les nom des fonctions avec le regex, les mettres dans une liste, ensuite appliquer la fonction de l'occurence
+  regex_FileNamePattern = re.compile(regex_FileName)
```

Comment:
Re-compiling static regexes on each call of `result` seems a bit inefficient to me. This could be done once at import-time.

Reply:
I am reading diff files line by line, just one time and for each line I am checking if I found occurrence with the regex. Could you explain more how to do regex once in the import time ?